src/eda_server/project.py

```diff
@@ -3,114 +3,305 @@
+        *cmd,
+        cwd=repo,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+        timeout=DEFAULT_TIMEOUT,
+    )
+    return result.stdout.strip()
 
-    finally:
-        pass
 
+async def git_archive(
+    repo: str,
+    commit_id: str,
+    *,
+    output: str,
+    format: Literal["tar", "tar.gz"] = "tar.gz",
+):
+    cmd = [GIT_BIN, "archive", "--format", format, "-o", output, commit_id]
+    await run_git_command(
+        *cmd,
+        cwd=repo,
+        stdout=asyncio.subprocess.DEVNULL,
+        stderr=asyncio.subprocess.PIPE,
+        timeout=GIT_ARCHIVE_TIMEOUT,
+    )
+
+
+async def clone_project(url: str, dest: str):
+    await git_clone(url, dest)
+    return await git_get_current_commit(dest)
+
+
+# ---- SYNC -------------------------------------------------------------------
+
+
+async def sync_project(db: AsyncSession, project_id: int, repo_dir: str):
+    for result in scan_project(repo_dir):
+        if result.kind is ProjectFileKind.RULEBOOK:
+            rulebook_id = await create_rulebook(
+                db,
+                project_id=project_id,
+                name=result.filename,
+                rulesets=result.raw_content,
+            )
+            await create_rules(db, rulebook_id, result.content)
+        elif result.kind is ProjectFileKind.PLAYBOOK:
+            await create_playbook(
+                db,
+                project_id=project_id,
+                name=result.filename,
+                playbook=result.raw_content,
+            )
+        elif result.kind is ProjectFileKind.INVENTORY:
+            await create_inventory(
+                db,
+                project_id=project_id,
+                name=result.filename,
+                inventory=result.raw_content,
+            )
+        elif result.kind is ProjectFileKind.EXTRA_VARS:
+            await create_extra_var(
+                db,
+                project_id=project_id,
+                name=result.filename,
+                extra_var=result.raw_content,
+            )
 
-# FIXME(cutwater): Remove try: .. finally: pass
-async def sync_project(project_id, large_data_id, tempdir, db: AsyncSession):
+
+IGNORED_DIRS = frozenset([".git", ".github"])
+
+
+class ProjectFileKind(Enum):
+    RULEBOOK = "rulebook"
+    INVENTORY = "inventory"
+    EXTRA_VARS = "extra_vars"
+    PLAYBOOK = "playbook"
+
+
+@dataclass
+class ScanResult:
+    kind: ProjectFileKind
+    filename: str
+    raw_content: str
+    content: Any
+
+
+def get_file_kind(content: Any) -> ProjectFileKind:
+    if is_rulebook_file(content):
+        return ProjectFileKind.RULEBOOK
+    elif is_inventory_file(content):
+        return ProjectFileKind.INVENTORY
+    elif is_playbook_file(content):
+        return ProjectFileKind.PLAYBOOK
+    else:
+        return ProjectFileKind.EXTRA_VARS
+
+
+def scan_file(path: str) -> Optional[ScanResult]:
+    # TODO(cutwater): Should we support both *.yml and *.yaml extensions ?
```

Comment:
We should, for sure. IMO. 
.yml is the most used, but .yaml is also widely adopted and is the recommended: https://yaml.org/faq.html